README.md

```diff
@@ -13,34 +13,48 @@ This plugin is available on [JetBrains Marketplace](https://plugins.jetbrains.co
 ![](docs/demo-workflow.gif)
 
 
-## Logging
+## How does it work
 
-SLF4J logging in this plugin has the following categories:
+Git Machete IntelliJ Plugin is a port of a great console tool - [git-machete](https://github.com/VirtusLab/git-machete#git-machete) - into an IntelliJ plugin.
+Git Machete can help you manage yours repository branches and keep it in sync with each other and its counterparts on remote repository.
+Let's see how this plugin can help you:
 
-```
-binding
-branchlayout
-gitcore
-gitmachete.backend
-gitmachete.frontend.actions
-gitmachete.frontend.externalsystem
-gitmachete.frontend.graph
-gitmachete.frontend.ui
-```
+### Where to find this plugin
 
-The standard practice in Java logging is to use fully-qualified name of each class as category;
-in our case, however, we're only ever using the categories provided above.
-FQCN (and method name), however, is always a part of the log message itself.
+Git Machete IntelliJ Plugin is available under `Git` section (in the bottom of the IDE) in the `Git Machete` tab.
+You can also use `Ctrl` + `Alt` + `Shift` + `M` shortcut to open this plugin.
 
-By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
-The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
-Tip: use `tail -f` to watch the log file as it grows.
+![](docs/open_git_machete.gif)
 
-To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
-A relatively small amount of `TRACE`-level logs is generated as well.
+### Machete file editor
 
+If you are familiar with console equivalent of this tool ([git-machete](https://github.com/VirtusLab/git-machete#git-machete)) then for sure you know what is the `machete` file.
+For those who don't know this yet, `machete` file is located inside `.git` directory in your repo.
+It describes relation between branches in your repository (these relations are probably determined by order of branch creation - which branch form which - but this is not a rule).
+For more information look [here](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02). <br/>
 
-## Development
+![](docs/machete_file_editor.gif)
+
+### Rebase - the heart of Git-Machete
```

Comment:
Don't use the hyphenated form, use `Git Machete` consistently instead, as per https://github.com/VirtusLab/git-machete-intellij-plugin/issues/104

Reply:
Also I'm reluctant to present rebase as the _heart_ of this software... I agree that as for now the plugin isn't of much use if someone doesn't want to use rebase, but still, it's better not to intimidate potential users who are prejudiced against rebase (and there's quite a few of them) :/

Reply:
git-machete CLI allows for using merge instead of rebase in pretty much all contexts (with the exception with `git machete reapply` where the former doesn't make much sense); we don't provide `merge parent into this branch` feature as for now, but it's relatively easy to add if/when requested (as happened with the CLI)